src/carbonio-files-ui-common/views/components/sharing/publicLink/PublicLink.test.tsx

```diff
@@ -162,145 +264,709 @@ describe.each<Node['__typename']>(['File', 'Folder'])('Public Link', (nodeType)
+			act(() => {
+				// run timers of modal
+				jest.runOnlyPendingTimers();
+			});
+			const modal = await screen.findByTestId('modal');
+			expect(
+				within(modal).getByText(RegExp(`${btnName} ${node.name} ${linkName}`, 'i'))
+			).toBeVisible();
+			const resolvedMsg = message.replace('{nodeName}', node.name);
+			expect(screen.getByText(resolvedMsg)).toBeVisible();
+		});
+
+		it('should close modal when click on X button', async () => {
+			const node = populateNode(nodeType);
+			const link = populateLink(node);
+			const mocks = {
+				Query: {
+					getLinks: mockGetLinks([link])
+				}
+			} satisfies Partial<Resolvers>;
+			const linkName = 'Link name';
+			const { user } = setup(
+				<PublicLink
+					nodeId={node.id}
+					nodeName={node.name}
+					linkTitle={'Link title'}
+					linkName={linkName}
+					linkDescription={'Link description'}
+				/>,
+				{ mocks }
+			);
+			await screen.findByText(link.url as string);
+			await user.click(
+				screen.getByRoleWithIcon('button', { name: /revoke/i, icon: ICON_REGEXP.revoke })
+			);
+			act(() => {
+				// run timers of modal
+				jest.runOnlyPendingTimers();
+			});
+			const closeButton = await screen.findByTestId(ICON_REGEXP.close);
+			expect(closeButton).toBeVisible();
+			await user.click(closeButton);
+			expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
+			expect(screen.queryByText(`Revoke ${node.name} ${linkName}`)).not.toBeInTheDocument();
+			expect(
+				screen.queryByText(
+					`By revoking this link, you are blocking access to ${node.name} for anyone who tries to use the link to access the item.`
+				)
+			).not.toBeInTheDocument();
+		});
+	});
+
+	describe('Delete link', () => {
+		it.each([
+			['Remove', getDayBefore(), ICON_REGEXP.deletePermanently],
+			['Revoke', null, ICON_REGEXP.revoke]
+		])('should delete link when click on %s button', async (btnName, expDate, icon) => {
+			const node = populateNode(nodeType);
+			const link = populateLink(node);
+			link.expires_at = expDate;
+			const mocks = {
+				Query: {
+					getLinks: mockGetLinks([link])
+				},
+				Mutation: {
+					deleteLinks: mockDeleteLinks([link.id])
+				}
+			} satisfies Partial<Resolvers>;
+			const linkName = 'Link name';
+			const { user } = setup(
+				<PublicLink
+					nodeId={node.id}
+					nodeName={node.name}
+					linkTitle={'Link title'}
+					linkName={linkName}
+					linkDescription={'Link description'}
+				/>,
+				{ mocks }
+			);
+			await screen.findByText(link.url as string);
+			await user.click(screen.getByRoleWithIcon('button', { name: btnName, icon }));
+			act(() => {
+				// run timers of modal
+				jest.runOnlyPendingTimers();
+			});
+			await screen.findByText(RegExp(`${btnName} ${node.name} ${linkName}`, 'i'));
+			const deleteButtons = screen.getAllByRole('button', { name: btnName });
+			expect(deleteButtons).toHaveLength(2);
+			await user.click(deleteButtons[1]);
```

Comment:
We should be able to get the single button by searching inside the modal (`within(modal)`)